src/main.rs

```diff
@@ -126,6 +126,18 @@ fn safe_main(program: &str, args: &[String]) -> Result<(), Error> {
     Ok(())
 }
 
+/// Flattens all causes of an error into a single string.
```

Comment:
This doesn't seem in line with failure's model. The top-level error is supposed to have enough information -- the remaining chain is for debugging or introspection.

Reply:
Yeah... I reached this approach from the long analysis in: https://github.com/rust-lang-nursery/failure/issues/189, which I understand to mean that annotating errors with some extra details is not something that failure can do well now.

Do you think that just "rewriting" the original IO error with the extra context, "by hand" as we did before with `map_err` is better?

Reply:
Bleh. The alternative is a lot of complex and noisy `with_context` calls. I suppose this will do.

Reply:
By the way: where do you get that the rest of the error chain is only for debugging? I've been re-reading the failure manual and I do not get that impression, especially by what's explained in the *Context* section under https://boats.gitlab.io/failure/fail.html. But there doesn't seem to be explicit guidance.